diff --color -Naur a/app/src/widgets/iconlabel.cpp b/app/src/widgets/iconlabel.cpp
--- a/app/src/widgets/iconlabel.cpp	2022-02-07 15:19:33.851222835 +0800
+++ b/app/src/widgets/iconlabel.cpp	2022-02-07 15:20:25.279867495 +0800
@@ -24,6 +24,7 @@
 #include <QLabel>
 #include <QIcon>
 #include <QPainter>
+#include <QPainterPath>
 
 IconLabel::IconLabel(QWidget *parent, bool wordWrap)
     : QWidget(parent)
